interpreter/cling/include/cling/Utils/ASTImportSource.h

```diff
@@ -0,0 +1,88 @@
+#ifndef CLING_ASTIMPORTSOURCE_H
+#define CLING_ASTIMPORTSOURCE_H
+
+#include "cling/Interpreter/Interpreter.h"
+
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Sema/Sema.h"
+#include "clang/AST/ASTContext.h"
+#include "clang/AST/ASTImporter.h"
+
+#include <string>
+#include <map>
+
+namespace clang {
+  class ASTContext;
+  class Expr;
+  class Decl;
+  class DeclContext;
+  class DeclarationName;
+  class GlobalDecl;
+  class FunctionDecl;
+  class IntegerLiteral;
+  class NamedDecl;
+  class NamespaceDecl;
+  class NestedNameSpecifier;
+  class QualType;
+  class Sema;
+  class TagDecl;
+  class TemplateDecl;
+  class Type;
+  class TypedefNameDecl;
+}
+
+namespace cling {
+  namespace utils {
+
+    class ASTImportSource : public clang::ExternalASTSource {
+
+      private:
+        cling::Interpreter *m_first_Interp;
+        cling::Interpreter *m_second_Interp;
+        const clang::TranslationUnitDecl *m_translationUnitI1;
+        const clang::TranslationUnitDecl *m_translationUnitI2;
+
+        clang::Sema *m_Sema;
+        /* A map for the DeclContext-to-DeclContext correspondence
+         * with DeclContexts pointers. */
+        std::map<const clang::DeclContext *, clang::DeclContext *> m_DeclContexts_map;
+        /* A map for all the imported Decls (Contexts). */
```

Comment:
Ditto: what is key, what is value (not their type but their meaning!) And what is this map used for?